### ISSUE TITLE
What is the difference between model_main.py and train.py in Tensorflow Object Detection API

### ISSUE THREAD

**Issue by csolfh** [tensorflow/models#7118]
What is the difference between model_main.py and train.py in Tensorflow Object Detection API

**Comment by QuickLearner171998** [comment]
model_main.py will run training and evaluation simultaneously.
train.py will do only training and for evaluation u have to run eval.py in parallel(in another terminal).
Also model_main.py has many bugs fixed that were in eval.py. 
So its better to use model_main.py

**Comment by bill2239** [comment]
I observed that model_main.py is 2x slower than train.py using the same config file and same machine. I am wondering if default settings are different. Maybe model_train.py use 32 bits and train.py use 16 bits? anyone?

**Comment by tombstone** [comment]
@csolfh train.py is deprecated. It depends on olds training loops. For TF1.15 we recommend using model_main.py / model_tpu_main.py.

**Comment by thagstrom123** [comment]
I am running both to compare with the same config file inception (pets). The loss rate is way way better using the train.py even if both run reasonably smooth, still trying to figure out why this is.
